formatter/formatSpecification.go

```diff
@@ -29,27 +29,47 @@ type formatter struct {
 	buffer bytes.Buffer
 }
 
+var nextItem gauge.Item
+
+func (formatter *formatter) SetNext(item gauge.Item) {
+	nextItem = item
+}
+
+func (formatter *formatter) GetNext() (item gauge.Item) {
+	return nextItem
+}
+
 func (formatter *formatter) Specification(specification *gauge.Specification) {
 }
 
-func (formatter *formatter) SpecHeading(specHeading *gauge.Heading) {
-	formatter.buffer.WriteString(FormatHeading(specHeading.Value, "="))
+func (formatter *formatter) Heading(heading *gauge.Heading) {
+	if heading.HeadingType == gauge.SpecHeading {
+		formatter.buffer.WriteString(FormatHeading(heading.Value, "="))
+	} else if heading.HeadingType == gauge.ScenarioHeading {
+		formatter.buffer.WriteString(FormatHeading(heading.Value, "-"))
+	}
 }
 
-func (formatter *formatter) SpecTags(tags *gauge.Tags) {
+func (formatter *formatter) Tags(tags *gauge.Tags) {
+	if !strings.HasSuffix(formatter.buffer.String(), "\n\n") {
+		formatter.buffer.WriteString("\n")
+	}
 	formatter.buffer.WriteString(FormatTags(tags))
+	if formatter.GetNext().Kind() != gauge.CommentKind || strings.TrimSpace(nextItem.(*gauge.Comment).Value) != "" {
```

Comment:
mixed use of `formatter.GetNext` and `nextItem`. Hence the proposal to have `nextItem` in `formatter`.